ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -244,11 +244,38 @@ function BlueprintService($log, $q, $sce, paletteApi, iconGenerator, dslService)
                 : paletteApi.getType(entity.type, entity.version, entity.config);
 
             promise.then((data)=> {
-                let sensors = getSensorsFromYaml(entity);
-                sensors.forEach((specificSensor) => {
-                    let sensor = specificSensor['brooklyn.config'];
-                    sensor.type = specificSensor['brooklyn.config'].targetType;
-                    sensor.sensorType = specificSensor.type;
+                getSensorsFromYaml(entity).forEach((specificSensor) => {
+                    const sensor = {
+                        sensorType: specificSensor.type,
+                        template: specificSensor,
+                        updateMetainfo: function() {
+                            this.name = this.template['brooklyn.config'].name;
+                            this.type = this.template['brooklyn.config'].targetType;
+                            this.description = this.template['brooklyn.config'].description;
+                        },
+                        deleteSensor: function() {
+                            let i = 0;
+                            const initializers = entity.metadata.get('brooklyn.initializers');
+                            while (i < initializers.length && initializers[i] != this.template) {
```

Comment:
I think I will have prefer a 
```
for(let i = 0; i < initializers.length; i++){ 
   if (initializers[i] === this.template) {
     initializers.splice(i, 1);
     break; // not necessary but without it it let think it would work if there is several element which is not the case
                // can be replaced by a "i--" also, that way it will work with several element.
   }
}
if (initializers.length == 0) {
    entity.metadata.delete('brooklyn.initializers');
}
```
(same for the two other loops)